cloud/scope/machine_pool.go

```diff
@@ -571,17 +628,313 @@ func (m *MachinePoolScope) TerminateInstancePool(ctx context.Context, instancePo
+			return core.IntelVmPlatformConfig{
+				IsSecureBootEnabled:            platformConfig.IntelVmPlatformConfig.IsSecureBootEnabled,
+				IsTrustedPlatformModuleEnabled: platformConfig.IntelVmPlatformConfig.IsTrustedPlatformModuleEnabled,
+				IsMeasuredBootEnabled:          platformConfig.IntelVmPlatformConfig.IsMeasuredBootEnabled,
+			}
+		case infrastructurev1beta1.PlatformConfigTypeIntelSkylakeBm:
+			return core.IntelSkylakeBmPlatformConfig{
+				IsSecureBootEnabled:            platformConfig.IntelSkylakeBmPlatformConfig.IsSecureBootEnabled,
+				IsTrustedPlatformModuleEnabled: platformConfig.IntelSkylakeBmPlatformConfig.IsTrustedPlatformModuleEnabled,
+				IsMeasuredBootEnabled:          platformConfig.IntelSkylakeBmPlatformConfig.IsMeasuredBootEnabled,
+			}
+		case infrastructurev1beta1.PlatformConfigTypeAmdMilanBm:
+			numaNodesPerSocket, _ := core.GetMappingAmdMilanBmPlatformConfigNumaNodesPerSocketEnum(string(platformConfig.AmdMilanBmPlatformConfig.NumaNodesPerSocket))
+			return core.AmdMilanBmPlatformConfig{
+				IsSecureBootEnabled:                      platformConfig.AmdMilanBmPlatformConfig.IsSecureBootEnabled,
+				IsTrustedPlatformModuleEnabled:           platformConfig.AmdMilanBmPlatformConfig.IsTrustedPlatformModuleEnabled,
+				IsMeasuredBootEnabled:                    platformConfig.AmdMilanBmPlatformConfig.IsMeasuredBootEnabled,
+				IsSymmetricMultiThreadingEnabled:         platformConfig.AmdMilanBmPlatformConfig.IsSymmetricMultiThreadingEnabled,
+				IsAccessControlServiceEnabled:            platformConfig.AmdMilanBmPlatformConfig.IsAccessControlServiceEnabled,
+				AreVirtualInstructionsEnabled:            platformConfig.AmdMilanBmPlatformConfig.AreVirtualInstructionsEnabled,
+				IsInputOutputMemoryManagementUnitEnabled: platformConfig.AmdMilanBmPlatformConfig.IsInputOutputMemoryManagementUnitEnabled,
+				PercentageOfCoresEnabled:                 platformConfig.AmdMilanBmPlatformConfig.PercentageOfCoresEnabled,
+				NumaNodesPerSocket:                       numaNodesPerSocket,
+			}
+		default:
+		}
+	}
+	return nil
+}
+
+func (m *MachinePoolScope) getWorkerMachineNSGs() []string {
+	instanceVnicConfiguration := m.OCIMachinePool.Spec.InstanceConfiguration.InstanceVnicConfiguration
+	if instanceVnicConfiguration != nil && len(instanceVnicConfiguration.NsgNames) > 0 {
+		nsgs := make([]string, 0)
+		for _, nsgName := range instanceVnicConfiguration.NsgNames {
+			for _, nsg := range m.OCICluster.Spec.NetworkSpec.Vcn.NetworkSecurityGroups {
+				if nsg.Name == nsgName {
+					nsgs = append(nsgs, *nsg.ID)
+				}
+			}
+		}
+		return nsgs
+	} else {
+		nsgs := make([]string, 0)
+		for _, nsg := range m.OCICluster.Spec.NetworkSpec.Vcn.NetworkSecurityGroups {
+			if nsg.Role == infrastructurev1beta1.WorkerRole {
+				nsgs = append(nsgs, *nsg.ID)
+			}
+		}
+		return nsgs
+	}
+}
+
+func (m *MachinePoolScope) GetInstanceConfiguration(ctx context.Context) (*core.InstanceConfiguration, error) {
```

Comment:
Please add the go doc comments for exported functions.

Reply:
fixed